cmd/gateway-gcs_test.go

```diff
@@ -181,3 +186,37 @@ func TestGCSMultipartDataName(t *testing.T) {
 		t.Errorf("expected: %s, got: %s", expected, got)
 	}
 }
+
+func TestFromMinioClientListBucketResultToV2Info(t *testing.T) {
+
+	// Input 1
+	listBucketResult := minio.ListBucketResult{
+		IsTruncated: false,
+		Marker:      "testMarker",
+		NextMarker:  "testMarker2",
+	}
+
+	listBucketV2Info := ListObjectsV2Info{
+		Prefixes:              make([]string, 0),
+		Objects:               make([]ObjectInfo, 0),
+		IsTruncated:           false,
+		ContinuationToken:     "testMarker",
+		NextContinuationToken: "testMarker2",
+	}
+
+	tests := []struct {
+		name   string
+		bucket string
+		result minio.ListBucketResult
+		want   ListObjectsV2Info
+	}{
+		{"test1", "testbucket1", listBucketResult, listBucketV2Info},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := fromMinioClientListBucketResultToV2Info(test.bucket, test.result); !reflect.DeepEqual(got, test.want) {
+				t.Errorf("fromMinioClientListBucketResultToV2Info() = %v, want %v", got, test.want)
+			}
+		})
+	}
```

Comment:
test case should test for prefixes and objects as well:

```
func TestFromMinioClientListBucketResultToV2Info(t *testing.T) {                                        
        objects :=                                                                                      
                                                                                                        
        listBucketResult := minio.ListBucketResult{                                                     
                IsTruncated: false,                                                                     
                Marker:      "testMarker",                                                              
                NextMarker:  "testMarker2",                                                             
                CommonPrefixes: []minio.CommonPrefix{{"one", "two"}},                                   
                Contents: []minio.ObjectInfo{{Key: "testobj"}}                                          
        }                                                                                               
                                                                                                        
        listBucketV2Info := ListObjectsV2Info{                                                          
                Prefixes:              []string{"one", "two"},                                          
                Objects:               []ObjectInfo{{Name: "testobj"}},                                 
                IsTruncated:           false,                                                           
                ContinuationToken:     "testMarker",                                                    
                NextContinuationToken: "testMarker2",                                                   
        }                                                                                               
                                                                                                        
        if got := fromMinioClientListBucketResultToV2Info("testbucket", listBucketResult); !reflect.Dee\
pEqual(got, listBucketV2Info) {                                                                         
                t.Errorf("fromMinioClientListBucketResultToV2Info() = %v, want %v", got, listBucketV2In\
fo)                                                                                                     
        }                                                                                               
}
```

commonPrefix inside minio-go is not exported, it should be exported first for this to work.